model/project_ref.go

```diff
@@ -101,7 +101,7 @@ type ProjectRef struct {
 	DisabledStatsCache    *bool    `bson:"disabled_stats_cache,omitempty" json:"disabled_stats_cache,omitempty"`
 
 	// List of commands
-	WorkstationConfig WorkstationConfig `bson:"workstation_config,omitempty" json:"workstation_config,omitempty"`
+	WorkstationConfig WorkstationConfig `bson:"workstation_config" json:"workstation_config"`
```

Comment:
I'm just curious about this change; did you remove this because it wasn't doing anything or because it was actively causing a problem or? I notice we do still have it for the others.

Reply:
Isn't that okay though? If WorkstationConfig is omitted entirely then GitClone / SetupCommands would implicitly be nil as well right? Or did that behavior also occur when setupCommands was explicitly empty array?

Reply:
I _think_ we included omitempty on all non-nil arrays to ensure that defaulting worked correctly, although that memory might not be quite right 🤔

Reply:
Yeah, the problem is that `omitempty` is being applied on both `nil` and `[]` :(

Reply:
Okay rip. Do other things also need to not be omitempty then? (TaskAnnotationSettings has a list inside it for example)

Reply:
No, SetupCommands is the only setting that is a nested, nullable list (i.e. one of the fields we flag).

Reply:
I'm afraid someone will break this on accident since it breaks convention; maybe add a comment to say why it doesn't have omitempty?